Sources/DolbyIORTSUIKit/Private/ViewModels/StreamViewModel.swift

```diff
@@ -287,7 +287,11 @@ final class StreamViewModel: ObservableObject {
             detailSourceAndViewRenderers: existingDetailSourceAndViewRenderers,
             settings: _
         ):
-            selectedVideoSource = currentlySelectedVideoSource
+            if sources.contains(currentlySelectedVideoSource) {
```

Comment:
Hi @CasperMok - Since source is a model object whose property values change quite frequently. I suggested do a comparison by `id` 

`selectedVideoSource = sources.first { $0.id ==  currentlySelectedVideoSource.id } ?? sortedSources[0]`

Reply:
Agree and done